src/main/java/com/lucas/plugintest/BanBlockCommand.java

```diff
@@ -24,39 +34,49 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
         Player player = (Player) sender;
 
         //testando a permissao
-        if(!player.hasPermission("plugintest.kick")){
-            player.sendMessage("Você não tem permissão");
+        if(!player.hasPermission("plugintest.ban")){
+            player.sendMessage(ChatColor.RED + "Você não tem permissão para executar esse comando!");
             return false;
         }
 
-
         //se argumentos forem menor que um, quer dizer que comando foi executado de forma incorreta
-        if(args.length < 1){
-            player.sendMessage("Uso correto: /ban <Jogador> <Motivo>");
+        if(args.length < 1 || args.length > 2){
+            player.sendMessage(ChatColor.RED + "Uso correto: /ban <Jogador> <Motivo>");
             return false;
         }
 
         //logica para pegar o destinatario do comando
-        Player target = player.getServer().getPlayer(args[0]);
+        Player target = Bukkit.getPlayer(args[0]);
 
         if(target == null){
-            player.sendMessage("Jogador não encontrado");
-            return false;
+            player.sendMessage(ChatColor.RED + "Jogador não encontrado!");
         }
 
-        String reason = "Sem motivo específicado";
-
+        UUID targetUUID = target.getUniqueId();
+        String reason = args.length > 1 ? String.join(" ", Arrays.copyOfRange(args, 1, args.length)) : "Sem motivo específico";
 
-        //se os arguments forem > 1, significa que a motivo, senao ele nao tera e subira a msg padrao da var
-        if(args.length > 1){
-           reason = String.join(" ", Arrays.copyOfRange(args, 1, args.length));
+        if(isPlayerBanned(targetUUID)){
+            player.sendMessage("O jogador já está banido!");
+            return false;
         }
 
-        target.getServer().getBanList(BanList.Type.NAME).addBan(target.getName(), reason, null, player.getName());
+        banLists.put(targetUUID, reason);
         target.kickPlayer(ChatColor.DARK_RED + "Você foi banido!" +  "\n Motivo: " + reason + "\nBanido por: " + player.getDisplayName());
-        player.sendMessage("Você baniu o jogador " + target.getDisplayName() + "Pelo motivo: " + reason);
+        player.sendMessage(ChatColor.RED + "Você baniu o jogador " + target);
 
         return true;
     }
 
+    public boolean isPlayerBanned(UUID playerUUID){
+        return banLists.containsKey(playerUUID);
```

Comment:
Maybe in the future we can implement a Player model to store player's data in a database, and something like "bool is_banned" to be checked here.

For now it is okay.

Reply:
of course, the hash and that stuff is just for realise tests in the workout